Actor.cs

```diff
@@ -0,0 +1,85 @@
+internal class Actor : MonoBehaviour
+{
+	private static List<Actor> _actors = new List<Actor>();
+	public int StartingHealth;
+	int _currentHealth;
+	private bool _isDead;
+	private CancellationTokenSource _healthCancellationTokenSource;
+	private CancellationTokenSource _deathCancellationTokenSource;
+
+	public event Action<Actor> ActorDied;
+	public event Action<Actor> ActorHealthChanged;
+
+	public void TakeDamage(int damage)
+	{
+		_currentHealth -= damage;
+		ActorHealthChanged?.Invoke(this);
+		RecordHealthToServer();
+		
+		if (_currentHealth <= 0)
+		{
+			_isDead = true;
+			ActorDied?.Invoke(this);
+			RecordDeathToServer();
+		}
+	}
+
+	private void Start()
+	{
+		_currentHealth = StartingHealth;
+	}
+
+	private void OnEnable()
+	{
+		_actors.Add(this);
```

Comment:
This should probably be in Start if we're tracking actor lifetime, chances are it'll get disabled from gameplay code for various things
If this is not the case & has been architected around that's fine though